### ISSUE TITLE
Is there a way to do in the mplot3d?

### ISSUE THREAD

**Issue by yu-yuxuan** [joferkington/mpldatacursor#27]
I find this works in 2D. Am I right?

**Comment by joferkington** [comment]
Yes, matplotlib's 3D plotting isn't fully supported.  

There are a number of reasons for this (e.g. matplotlib doesn't actually do 3D plotting, so it's very difficult to tell what's "in front"), but I'm hoping to figure out a work-around soon.  It's a very frequently requested feature.

**Comment by tacaswell** [comment]
attn @weathergod

**Comment by WeatherGod** [comment]
mplot3d has a function for guessing a 3d coordinate from a 2d mouse coordinate. That is how it is able to display the format_coord() information. In axes3d.py:
```
    def format_coord(self, xd, yd):
        """
        Given the 2D view coordinates attempt to guess a 3D coordinate.
        Looks for the nearest edge to the point and then assumes that
        the point is at the same z location as the nearest point on the edge.
        """

        if self.M is None:
            return ''

        if self.button_pressed in self._rotate_btn:
            return 'azimuth=%d deg, elevation=%d deg ' % (self.azim, self.elev)
            # ignore xd and yd and display angles instead

        p = (xd, yd)
        edges = self.tunit_edges()
        #lines = [proj3d.line2d(p0,p1) for (p0,p1) in edges]
        ldists = [(proj3d.line2d_seg_dist(p0, p1, p), i) for \
                i, (p0, p1) in enumerate(edges)]
        ldists.sort()
        # nearest edge
        edgei = ldists[0][1]

        p0, p1 = edges[edgei]

        # scale the z value to match
        x0, y0, z0 = p0
        x1, y1, z1 = p1
        d0 = np.hypot(x0-xd, y0-yd)
        d1 = np.hypot(x1-xd, y1-yd)
        dt = d0+d1
        z = d1/dt * z0 + d0/dt * z1

        x, y, z = proj3d.inv_transform(xd, yd, z, self.M)

        xs = self.format_xdata(x)
        ys = self.format_ydata(y)
        zs = self.format_zdata(z)
        return 'x=%s, y=%s, z=%s' % (xs, ys, zs)
```

Maybe this will help?

**Comment by joferkington** [comment]
@WeatherGod - Yes, absolutely! I was actually looking for exactly that.  Don't know why I didn't come across it earlier!  (I'll blame it on distraction... I'm too focused on work lately.)

**Comment by joferkington** [comment]
Implemented in 30e1d068726fcd938307b426e314a24ee85ca91c.

Sorry for the delay!
